lib/ext.Infobox.js

```diff
@@ -0,0 +1,32 @@
+var Util = require( './mediawiki.Util.js' ).Util,
+	coreutil = require('util'),
+	ExtensionHandler = require('./ext.core.ExtensionHandler.js').ExtensionHandler,
+	defines = require('./mediawiki.parser.defines.js'),
+	InfoboxRequest = require( './mediawiki.ApiRequest.js' ).InfoboxRequest;
+
+var Infobox = function() {};
+
+// Inherit functionality from ExtensionHandler
+coreutil.inherits(Infobox, ExtensionHandler);
+
+Infobox.prototype.handleInfobox = function(manager, pipelineOpts, token, cb) {
+	var args = manager.frame.parentFrame.templateArgInfo.dict.params;
+	for ( var arg in args ) {
+		args[arg] = args[arg].wt;
+	}
+
+	// This async signalling has to happen before any further pipeline processing
+	cb({'async': true});
+
+	(new InfoboxRequest(manager.env, token.getAttribute('source'), args, manager.env.page.name)).once('src', function(error, html) {
```

Comment:
No. We don't really need to pass it as separated argument but given that *Request.js classes are just simple wrappers for MW api.php methods it makes sense to give flexibility of deciding what the values for parameters are to user of those classes (simple wrappers).